# Undertanding the IRIS data set

The notebook aims to undertand the content of the IRIS data set.

- Emma Gabriela Alfaro de la Rocha
- A01740229


In [28]:
ls "/Users/emmagalfaro/Documents/GitHub/TC1002S/NotebooksProfessor/datasets/iris/iris.csv"

/Users/emmagalfaro/Documents/GitHub/TC1002S/NotebooksProfessor/datasets/iris/iris.csv


In [29]:
# Import the packages that we will be using
import matplotlib.pyplot as pltRuta
import pandas as pd

colnames = ['sepal_length', 'sepal_width', 'petal_length', 'petal_width', 'flowerType']
# Dataset url
url = "/Users/emmagalfaro/Documents/GitHub/TC1002S/NotebooksProfessor/datasets/iris/iris.csv"
# Load the dataset
dataset = pd.read_csv(url, header=None, names=colnames )


In [30]:
# Print the dataset
dataset

sepal_length  sepal_width  petal_length  petal_width      flowerType
0             5.1          3.5           1.4          0.2     Iris-setosa
1             4.9          3.0           1.4          0.2     Iris-setosa
2             4.7          3.2           1.3          0.2     Iris-setosa
3             4.6          3.1           1.5          0.2     Iris-setosa
4             5.0          3.6           1.4          0.2     Iris-setosa
..            ...          ...           ...          ...             ...
145           6.7          3.0           5.2          2.3  Iris-virginica
146           6.3          2.5           5.0          1.9  Iris-virginica
147           6.5          3.0           5.2          2.0  Iris-virginica
148           6.2          3.4           5.4          2.3  Iris-virginica
149           5.9          3.0           5.1          1.8  Iris-virginica

[150 rows x 5 columns]

In [31]:
# Print the number of rows
Nrows = dataset.shape[0]
Nrows

150

In [32]:
# Print the number of columns
Ncols = dataset.shape[1]
Ncols

5

In [33]:
#How many observations are for each type of flower?
observations = dataset['flowerType'].value_counts()
print(f"Number of observations for each type of flower:\n{observations}")

Number of observations for each type of flower:
Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
Name: flowerType, dtype: int64


# Data types

In [34]:
dataset.dtypes

sepal_length    float64
sepal_width     float64
petal_length    float64
petal_width     float64
flowerType       object
dtype: object

# Activity: work with the iris dataset

1. Load the iris.csv file in your computer and understand the dataset


2. How many observations (rows) are in total?
    150
    
3. How many variables (columns) are in total? What do they represent?
    5

4. How many observations are for each type of flower?
    Number of observations for each type of flower:
    Iris-setosa        50
    Iris-versicolor    50
    Iris-virginica     50

5. What is the type of data for each variable?
    sepal_length    float
    sepal_width     float
    petal_length    float
    petal_width     float
    flowerType      object

6. What are the units of each variable?
    sepal_length    cm
    sepal_width     cm
    petal_length    cm
    petal_width     cm
    flowerType      N/A